.github/workflows/codecov.yaml

```diff
@@ -0,0 +1,163 @@
+        run: |
+          cmake --version
+          ${CLANG_ROOT_DIR}/bin/clang --version
+          gcc --version
+          ls -la
+      - name: Prepare ccache keys
+        id: ccache_cache_prop
+        shell: cmake -P {0}
+        run: |
+          string(REPLACE "/" "_" repo ${{github.repository}})
+          string(REPLACE "-" "_" repo2 ${repo})
+          message("::set-output name=repo::${repo2}")
+
+      - name: use cache
+        id: cache-ccache
+        uses: nashif/action-s3-cache@master
+        with:
+          key: ${{ steps.ccache_cache_prop.outputs.repo }}-${{github.event_name}}-${{matrix.platform}}-codecov-ccache
+          path: /github/home/.ccache
+          aws-s3-bucket: ccache.zephyrproject.org
+          aws-access-key-id: ${{ secrets.CCACHE_S3_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.CCACHE_S3_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+
+      - name: ccache stats initial
+        run: |
+          test -d github/home/.ccache && mv github/home/.ccache /github/home/.ccache
+          ccache -M 10G -s
+
+      - name: Run Tests with Twister (Push)
+        continue-on-error: true
+        run: |
+          export ZEPHYR_BASE=${PWD}
+          export ZEPHYR_TOOLCHAIN_VARIANT=zephyr
+          mkdir -p coverage/reports
+          ./scripts/twister -N -v --filter runnable -p ${{ matrix.platform }} --coverage -T tests
+
+      - name: Generate Coverage Report
+        run: |
+          mv twister-out/coverage.info lcov.pre.info
+          lcov -q --remove lcov.pre.info mylib.c --remove lcov.pre.info tests/\* \
+            --remove lcov.pre.info samples/\* --remove lcov.pre.info ext/\* \
+            --remove lcov.pre.info *generated*  \
+            -o coverage/reports/${{ matrix.platform }}.info --rc lcov_branch_coverage=1
+
+      - name: ccache stats post
+        run: |
+          ccache -s
+
+      - name: Upload Coverage Results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: Coverage Data (Subset ${{ matrix.platform }})
+          path: coverage/reports/${{ matrix.platform }}.info
+
+  publish-coverage-results:
+    name: "Publish Coverage Results"
+    needs: codecov
+    runs-on: ubuntu-latest
+      # the codecov job might be skipped, we don't need to run this job then
+    if: success() || failure()
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Download Artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: coverage/reports
+
+      - name: Move coverage files
+        run: |
+          mv ./coverage/reports/*/*.info ./coverage/reports
+          ls -la ./coverage/reports
+
+      - name: Generate list of coverage files
+        id: get-coverage-files
+        shell: cmake -P {0}
+        run: |
+          file(GLOB INPUT_FILES_LIST  "coverage/reports/*.info")
+          SET(MERGELIST "")
```

Comment:
The usual zephyr style is to use lowercase for all cmake function calls, etc.; why is this using `SET` instead of `set`, `FOREACH` instead of `foreach`, and so on?

Reply:
because this is not a zephyr cmake file really
I think I copied part of that from somewhere. Anyways, will fix